interview-questions.md

```diff
@@ -8,42 +8,88 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is object-oriented programming? How is it different than functional programming?
 
-Your answer:
+Your answer: Object-oriented programming is a development structure that forms code and functionality around using objects. This can be seen evidently in the lanuguage Ruby where everthing is essentialy an object. 
 
-Researched answer:
+Researched answer: a softeware development model that structures design around objects as opposed to functions and logic. An object is data that has unique behavior and its own attributes. 
 
-1. What is the difference between a Float and an Integer in Ruby?
+answers via: https://www.techtarget.com/searchapparchitecture/definition/object-oriented-programming-OOP#:~:text=Object%2Doriented%20programming%20(OOP)%20is%20a%20computer%20programming%20model,has%20unique%20attributes%20and%20behavior.
 
-Your answer:
 
-Researched answer:
+2. What is the difference between a Float and an Integer in Ruby?
 
-1. Ruby has an implicit return. What does that mean?
+Your answer: In Ruby there are two ways on desplaying a number. Float is the keyword that allows users to utilize decimal points. Integar is the keyword that refers to whole numbers. It will not give decimals and is the same as the number keyword, but integer is for the language Ruby. 
 
-Your answer:
+Researched answer: In ruby float is a number that contains the use of decimals. Any number without a decimal is a integer in Ruby. For example the number 24 is a integer, however the number 24.0 is a float. 
 
-Researched answer:
+answers via: https://www.digitalocean.com/community/tutorials/understanding-data-types-in-ruby
 
-1. What is a block in Ruby?
 
-Your answer:
+3. Ruby has an implicit return. What does that mean?
 
-Researched answer:
+Your answer: The language Ruby is written in a way that is simple to read. Implicit return is the languages way of returning the last line of code within a method without using any keywords. 
 
-1. How do you extract a value in a Ruby hash?
+Researched answer: Ruby will return the last line of a methods code. This is called the implicit return. In the example below we have a method called additon that has one parameter. inside the method we have a new variable called total which is holding num + num. On the next line we have 'total' which will automaticaly be the return because of Ruby implicit return. 
 
-Your answer:
+    def addition (num)
+        let total = num + num 
+        total
+    end
```

Comment:
great! And in fact you could make this even shorter by using the implicit return on your code action.
```
    def addition (num)
        num + num 
    end
 ```